### ISSUE TITLE
Vertical tab disappears dragging between windows

### ISSUE THREAD

**Issue by rillian** [brave/brave-browser#30389]
## Description 

Testing vertical tabs on Linux I found it's possible for the tab icon to disappear when dragging it between windows.

This is  on Gnome Desktop, Fedora 38, x86_64, focus-on-hover enabled in gnome-tweaks.

## Steps to Reproduce

   1. Enable vertical tabs
   2. Open two windows
   3. Press super-leftarrow and super-rightarrow to dock the two windows side-by-side covering the full display
   4. Drag a tab from one window to the tab area of the other
   5. Wiggle it in and out of the second window

## Actual result:

Tab icon/detached window disappears. Releasing the mouse either returns the tab to the original window, or leaves it in a third window hidden behind the other two.

[Demonstration of the issue](https://github.com/brave/brave-browser/assets/225102/9490ff1b-f9e9-4103-824d-5630b59c86cf)

## Expected result:
Tab should say connected to the pointer until it's released.

## Reproduces how often: 

So far I can do it reliably

## Brave version (brave://version info)

1.52.92 Chromium: 113.0.5672.92 (Official Build) beta (64-bit)

## Version/Channel Information:

- Can you reproduce this issue with the current release? No vertical tabs there, I think?
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

Follow-up to testing requests on Linux for https://github.com/brave/brave-browser/issues/29993

**Comment by rebron** [comment]
cc: @sangwoo108 @fallaciousreasoning

**Comment by morisono** [comment]
This bug is so **CRITICAL**.

**Comment by fallaciousreasoning** [comment]
cc @simonhong

**Comment by simonhong** [comment]
> cc [@simonhong](https://github.com/simonhong) did you recently fix this one?

Yeah, it's fixed by https://github.com/brave/brave-core/pull/27971
It's applied to latest stable `1.77.x`

**Comment by simonhong** [comment]
Duplicated with https://github.com/brave/brave-browser/issues/44397
